stubs/AclProviderInterface.stub

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Symfony\Component\Security\Acl\Model;
+
+use Symfony\Component\Security\Acl\Model\AclInterface;
+
+/**
+ * @template TAcl of AclInterface
+ */
+interface AclProviderInterface extends AclProviderInterface
```

Comment:
`AclProviderInterface extends AclProviderInterface`?

Reply:
And BTW you don't have to solve this with generics, I'm not sure it makes sense here. You can simply override the method in the child MutableAclProviderInterface and say it returns MutableAclInterface.

Reply:
> And BTW you don't have to solve this with generics, I'm not sure it makes sense here. You can simply override the method in the child MutableAclProviderInterface and say it returns MutableAclInterface.

I made a new try.

Reply:
Yeah, this is nicer, thanks. Please fix the build, you need to follow this tidbit:

> Why is PHPStan complaining about “Class not found” in a stub file?

https://phpstan.org/user-guide/stub-files

Reply:
Yeah, namespaces could be honored in the directory structure :)

For the directories it'd have to be a new config key `stubDirectories` but I don't think it's really necessary.

Reply:
I made the following changes:
- honored namespaces with directory inside the stubs one
- run the linter on the stubs files.
- Add the missing stubs classes.

Not every methods are defined in a stubs file. In the same way, do I have to write `interface MutableAclProviderInterface extends AclProviderInterface` in a stub file or can I just write 
`interface MutableAclProviderInterface` and avoid defining `AclProviderInterface` ?

Reply:
The build is still failing. It's better to write `interface MutableAclProviderInterface extends AclProviderInterface`, seems more complete, although that information is currently unused from the stubs.